### ISSUE TITLE
x11grab unable to fetch screen

### ISSUE THREAD

**Issue by yarons** [muammar/mkchromecast#184]
[x11grab @ 0x55a9e17ebc80] Cannot open display :0.0+0,0, error 1.
    :0.0+0,0: Input/output error

Full command was:

    mkchromecast --screencast --video

OS: Manjaro Linux 17.1.18
Tried with `sudo`, no change.

Any clue?

**Comment by yarons** [comment]
Well, I have a way to bypass this error.

Apparently Mkchromecast is ignoring the current display, instead the parameters to ffmpeg are hardcoded.

https://github.com/muammar/mkchromecast/blob/3b4817c55f700bba438e741871b5b221f95e1ad3/mkchromecast/video.py#L78

This should either be determined automagically or some parameter should be added, otherwise if your screen is not on `DISPLAY=:0` will get some red error and the cast will break.

In order to determine what is my `DISPLAY` value I typed in `echo $DISPLAY` in terminal and I wrote the value I found there into the `mkchromecast/video.py` file in the relevant line.

Anyways, it's working now, hoping we can get some additional parameter for that (or auto detection if possible).
Thanks!

**Comment by mjozefcz** [comment]
Hey

same issue here. For now I hot-fixed it by hardcoding my screen.

```
diff --git a/mkchromecast/video.py b/mkchromecast/video.py
index 63e9ad2..355bf59 100644
--- a/mkchromecast/video.py
+++ b/mkchromecast/video.py
@@ -103,7 +103,7 @@ elif screencast is True:
         "-s",
         screen_size,
         "-i",
-        "{}.0+0,0".format(display),
+        ":0.0",
         "-vcodec",
         vcodec,
     ]
```

**Comment by yarons** [comment]
Have you tried changing the dot to a colon?
What could "display" be? I think that the notation should include colon somehow anyway.

**Comment by mathbr** [comment]
@mjozefcz Thanks for the fix, I did suggest this in #354 now.

`DISPLAY` [is of the format](https://askubuntu.com/a/432257) `:D.S`:

* D is a sequence number (usually 0). It can be varied if there are multiple displays connected to one computer.
* S is the screen number. A display can actually have multiple screens. Usually there's only one screen though where 0 is the default.
